### DISCUSSION TITLE
Whatsapp scrapes Meta tags up to 300KB of page source

### DISCUSSION THREAD

**Discussion by Asko-Dev** [nuxt/nuxt#29650] [Questions]
Hi, I am using Nuxt3 with vuetify ... maybe my setup is wrong, I simply have latest vuetify installed and mount it through defineNuxtPlugin, this is my nuxt-config with potentially relevant options

```
css: [
        "vuetify/styles",
        "@/assets/css/main.scss", // main global styles
    ],
    build: {
        transpile: ["vuetify"], // Ensure Vuetify is transpiled
    },
    plugins: [
        "@/plugins/index.js", // Ensure this line points to plugins directory
    ],
    app: {
        head: {
            meta: [{ charset: "utf-8" }],
            link: [{ rel: "icon", type: "image/svg+xml", href: "/favicon.svg" }],
            htmlAttrs: {
                lang: "cs",
            },
        },
    },
    modules: [
        "@pinia/nuxt",
        "@pinia-plugin-persistedstate/nuxt",
        "@nuxtjs/google-fonts",
        "@nuxtjs/device",
        "@zadigetvoltaire/nuxt-gtm",
        "@nuxtjs/sitemap",
        "@nuxtjs/robots",
        "dayjs-nuxt",
    ],
    vite: {
        ssr: {
            noExternal: ["vuetify"],
        },
        css: {
            preprocessorOptions: {
                scss: {
                    api: "modern-compiler", 
                },
            },
        },
        optimizeDeps: {
            include: ["vuetify"], // Specify libraries to pre-bundle
        },
    }
```
    
The thing is that the page source of my page loads the entire css of vuetify and nuxt bundles ... only then it loads meta tags, example here

```
<!DOCTYPE html><html  lang="cs" data-capo=""><head><meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<title>Recept Nejjednodušší jogurtový dort | NaGram.cz</title>
<link rel="preconnect" href="https://ik.imagekit.io/">
<style id="vuetify-theme-stylesheet">:root {
      --v-theme-background: 255,255,255;
      --v-theme-background-overlay-multiplier: 1;
      --v-theme-surface: 255,255,255;
      --v-theme-surface-overlay-multiplier: 1;
      --v-theme-surface-bright: 255,255,255;
      --v-theme-surface-bright-overlay-multiplier: 1;
(...)
```

all other meta tags I have in `<Head>` on the page go under this. And they always go under I tried putting the tags in nuxt-config, I tried using definePageMeta, I tried useHead ... always go under. 

It wouldn't bother me too much but Whatsapp came with a [300KB rule](https://developers.facebook.com/docs/whatsapp/link-previews/),  they display meta tags found up until 300KB of page source. Those stylesheets and all unfortunately take up as much and the only tag that gets displayed is therefore <title> ... no image, no description, nothing.

Is there a way to reduce the loaded css? Or at least push the meta tags up before the stylesheets and nuxt modules? 

Thank you

**Comment by rendisimamora** [comment]
You can use Nuxt's server-side middleware to insert meta tags at the top of the HTML. This ensures essential tags, like Open Graph data, load before CSS or other inline styles.

`In nuxt.config.js`

```
export default {
  render: {
    csp: true,
    asyncScripts: true,
    injectScripts: false
  }
}
```

and you can use purgecss to remove unused css

**Reply by Asko-Dev** [reply]
I unfortunately couldn't use this as I am using nuxt3, but in the end I found out you can just add `tagPriority: "critical" ` to a tag and it pushes it up. 

```useHead({
    title: `Recept ${recipe.name} | NaGram.cz`,
    meta: [
        { name: "description", content: description, tagPriority: "critical" },
(...)
 ],
})
```
